### PR TITLE
feat: aws secret manager certificate

### PR DESCRIPTION
**Pull request by mstasgravitee**
## :id: Reference related issue. 
https://gravitee.atlassian.net/browse/AM-3308

## :man: :woman:@mentions of the person or team responsible for reviewing proposed changes

This is a nice example: https://github.com/gravitee-io/gravitee-access-management/pull/1822

### CODE REVIEW

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java [hunk 1]
```diff
@@ -244,97 +229,122 @@ public Single<Certificate> create(String domain, NewCertificate newCertificate,
...
-                                    .filter(map -> map.getValue().getWidget() != null && "file".equals(map.getValue().getWidget()))
-                                    .map(Map.Entry::getKey)
-                                    .findFirst().orElse(null);
-                            String oldFileInformation = oldCertificateConfiguration.get(key).asText();
-                            String fileInformation = certificateConfiguration.get(key).asText();
-                            // file has changed, let's update it
-                            if (!oldFileInformation.equals(fileInformation)) {
-                                JsonNode file = objectMapper.readTree(certificateConfiguration.get(key).asText());
-                                byte[] data = Base64.getDecoder().decode(file.get(CONTENT).asText());
-                                certificateToUpdate.setMetadata(Collections.singletonMap(CertificateMetadata.FILE, data));
```
Comment (mstasgravitee) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java:304]: I am not sure how it has worked with immutable map

In AbstractCertificateProvider is executed
  // update metadata
certificateMetadata.getMetadata().put(CertificateMetadata.DIGEST_ALGORITHM_NAME, signature.getDigestName());

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java [hunk 2]
```diff
@@ -118,9 +119,14 @@ public class CertificateServiceImpl implements CertificateService {
...
     private static final String CONTENT = "content";
-    private static final String STORE_PASS = "storepass";
     private static final String ALIAS = "alias";
     private static final String NAME = "name";
+    private static final String AWS_AM_CERTIFICATE = "aws-am-certificate";
+    private static final String SECRET_NAME = "secretname";
+    private static final String KEY_PASS = "keypass";
+    private static final String STORE_PASS = "storepass";
+    private static final String CACHE = "cache";
+    private static final String REGION = "region";
```
Comment (mstasgravitee) [inline gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java:129]: cache and region are required. To test it I have to push changes to the plugin.

File: gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java (hunk 1 above)
Comment (leleueri) [reply gravitee-am-service/src/main/java/io/gravitee/am/service/impl/CertificateServiceImpl.java:304]: Indeed, this is strange but I'm ok with your change to move it to non singleton HashMap

**Review by leleueri — COMMENTED** [review]
LGTM, we just need to make the dependency accessible

**Review by leleueri — APPROVED** [review]

**Pull request merged**